### ISSUE TITLE
the epoch_IoU of retrained refinement network can only up to 0.35 on deepglobe dataset

### ISSUE THREAD

**Issue by DwRolin** [VinAIResearch/MagNet#18]
I tried to retrain the segmentation backbone and refinement network following the guideline in readme https://github.com/VinAIResearch/MagNet#training-backbone-networks.  
The best_mIoU of retrained backbone fpn is 0.6363 , this result is close to the baseline IoU 0.6722 shown in readme.
![image](https://user-images.githubusercontent.com/34975133/171681919-7685bfa8-be0d-4e22-8328-d40cdc6c406d.png)
In this sense, the performance of retrained refinement network with retrained backbone should be close to the performance with pretrained backbone.
In the retraining of refinement network,  the change of  epoch_IoU  with pretrained backbone was like following image, 
![image1](https://user-images.githubusercontent.com/34975133/171682075-5101cbfb-ec03-416d-b3b3-f2d9d48016e5.png)
the change of  epoch_IoU  with retrained backbone was like following image. 
![image2](https://user-images.githubusercontent.com/34975133/171682192-4680f75d-9a3f-4be3-9d32-ba131b496c4c.png)
With the retrained backbone, the epoch_IoU can only up to 0.35. 
I tried to find the difference between pretrained backbone and retrained backbone.
I separated the validate part from backbone/[train.py](http://train.py/) to evaluate the performance of pretrained backbone. https://github.com/DwRolin/temp_code/blob/main/eval_pretrain.py
What's strange is that the MeanIU of pretrained backbone is only 0.07.
I would like to know what causes this contradiction and how to make the retrained refinement network work well.

**Comment by DwRolin** [comment]
The following is the training log for the backbone fpn.
[resnet_fpn_train_612x612_sgd_lr1e-2_wd5e-4_bs_12_epoch484_2022-05-05-16-57_train.log](https://github.com/VinAIResearch/MagNet/files/8825936/resnet_fpn_train_612x612_sgd_lr1e-2_wd5e-4_bs_12_epoch484_2022-05-05-16-57_train.log)

**Comment by hmchuong** [comment]
Hi, 
Are you working on the DeepGlobe database? Can you check that the [CUDNN. ENABLED](https://github.com/VinAIResearch/MagNet/blob/main/backbone/experiments/deepglobe/resnet_fpn_train_612x612_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml#L4) are the same in both backbone and your refinement training script? and it should be `True`.

**Comment by DwRolin** [comment]
Thank you for your serious reply!
I set [CUDNN.ENABLED](https://github.com/VinAIResearch/MagNet/blob/main/backbone/experiments/deepglobe/resnet_fpn_train_612x612_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml#L4) to `true`, then the issue is resolved. 
I cloned this code about six months ago, at that time, the CUDNN.ENABLED was set to `false`.
And I am curious about why the CUDNN.ENABLE is set to `false `in hrnet_ocr_w18_train_256x128_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml, while it is set to `true` in resnet_fpn_train_612x612_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml.

**Comment by hmchuong** [comment]
Hi,

The reason is that the Batchnorm2d behavior depends on CUDNN
